pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -153,42 +153,21 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
 			newCRB := newClusterRoleBinding(clusterRoleBindingName)
 			err := r.client.Create(context.TODO(), newCRB)
 			if err != nil {
-				groupPermissionConditions := instance.Status.Conditions
-
-				// make new condition
-				newCondition := managedv1alpha1.Condition{
-					LastTransitionTime: metav1.Now(),
-					ClusterRoleName:    clusterRoleName,
-					Message:            "Unable to create ClusterRoleBinding: <error from platform>",
-					Status:             true,
-					State:              "Failed",
-				}
-
-				// update GroupPermission object with new conditions
-				instance.Status.Conditions = append(groupPermissionConditions, newCondition)
+				// calls on helper function to update the condition of the groupPermission object
+				groupPermissionObject := updateCondition(instance, "Unable to create ClusterRoleBinding: <error from platform>", clusterRoleName, true, "Failed")
 
-				err = r.client.Status().Update(context.TODO(), instance)
+				err = r.client.Status().Update(context.TODO(), groupPermissionObject)
 				if err != nil {
 					reqLogger.Error(err, "Failed to update condition.")
 					return reconcile.Result{}, err
 				}
 				reqLogger.Error(err, "Failed to create clusterRoleBinding")
 				return reconcile.Result{}, err
 			} else {
-				groupPermissionConditions := instance.Status.Conditions
-
-				// make new condition
-				newCondition := managedv1alpha1.Condition{
-					LastTransitionTime: metav1.Now(),
-					ClusterRoleName:    clusterRoleName,
-					Status:             true,
-					State:              "Created",
-				}
 
-				// update GroupPermission object with new conditions
-				instance.Status.Conditions = append(groupPermissionConditions, newCondition)
+				groupPermissionObject := updateCondition(instance, "", clusterRoleName, true, "Created")
```

Comment:
Should a success have a message?

Reply:
added "successfully created ClusterRoleBinding" as the message